modules/@apostrophecms/util/ui/src/http.js

```diff
@@ -153,9 +153,16 @@ export default () => {
     let keys;
     let i;
 
-    if (options.qs) {
-      url = apos.http.addQueryToUrl(url, options.qs);
-    }
+    qat = url.indexOf('?');
+    const qs = {
+      ...qat !== -1 && apos.http.parseQuery(url.substring(qat)),
+      ...options.qs || {},
+      ...options.qs?.aposLocale
+        ? { aposLocale: options.qs.aposLocale }
+        : { aposLocale: apos.getActiveLocale() }
+    };
+    url = apos.http.addQueryToUrl(url, qs);
```

Comment:
This is part of the tech design.
It should be since when not in a localization mode we simply pass the current locale which does not change the behavior.